src/Microsoft.ML.Data/DataView/CacheDataView.cs

```diff
@@ -193,12 +193,15 @@ public int MapInputToCacheColumnIndex(int inputIndex)
 
         public Schema Schema => _subsetInput.Schema;
 
-        public long? GetRowCount(bool lazy = true)
+        /// <summary>
+        /// This implementation can cost more than O(1) operations because it tries its best to
+        /// compute the actual row count.
+        /// </summary>
+        /// <returns>number of rows</returns>
+        public long? GetRowCount()
         {
             if (_rowCount < 0)
             {
```

Comment:
>{ [](start = 12, length = 1)

I suggest to return `null` here. Look at the comment for `GetRowCount()`, it promises to return null sometimes and then start returning non-null

Reply:
Sure.